### ISSUE TITLE
Create test mode that moves motors and checks encoders for any system with motors and encoders

### ISSUE THREAD

**Issue by sjcbulldog** [errorcodexero/newarch#93]

**Comment by 1425** [comment]
If you don't already have it, you might want to also check that the current use makes sense as well.  

This is of course reported from the PDB for all the motors but may be available from the motor controller as well depending on what type it is.

**Comment by sjcbulldog** [comment]
I assume the 1425 account is Eric?  We have looked at current for motors a little in the past.  We can add a zero current detect and will make that part of the self test as we do them.  What we will do is add the capability to the framework first, and then implement the actual tests based on time and schedule.  I would like to attack drivebase first because I can count at least 10 matches across the whole season where we lost auto modes because drivebase encoders were not working.

I would like to do something more with motors like be able to detect if we are operating them outside of the range expected.  This means getting a normal operating range for the motor.  I have not had luck getting this from mechanical and electrical in the past.  Maybe we just put characterization code in that we use while developing and then empirically set these limits.  Anyway, step one is straight forward and we are shooting to have this in bunny bots.

**Comment by 1425** [comment]
Yes, this is Eric.  In the past, currents thresholds have been set empirically.  Expected values are pretty dependent on what you're telling them to do - I'm not sure how complex a formula you'd hope for.  It would basically be a combination of voltage applied + what things it's touching + momentum.
